### PR TITLE
ASC-1031 Run system tests based on job scenario

### PR DESCRIPTION
**Pull request by johnduarte**
This commit parses the `RE_JOB_SCENARIO` to determine additional system
tests to execute based on job context.  The testing scenario is taken
from the last segment of the `RE_JOB_SCENARIO` with any preceding
numerical prefix stripped from the segment.  For example, a value of
`3ctlr_2comp_3ceph` for the `RE_JOB_SCENARIO` will be parsed into the
value of `ceph` for the `SYS_JOB_SCENARIO` variable.  This value is then
appended to the list of system test molecule scenarios to execute for
the system-test job run.

This update allows system test for incompatible scenarios to
introspected and tested in context of the given job. For example, `ceph`
and `swift` scenarios can be run in different deployment jobs and the
undefined scenario will not be run.

### CODE REVIEW

**Review by ryan-rs — APPROVED** [review]

File: gating/periodic/run_system_tests.sh
Comment (coreywright) [inline gating/periodic/run_system_tests.sh]: Just inquiring about the reason for `pushd` as i don't see any use of a corresponding `popd`.

**Comment by coreywright** [comment]
left a comment out of curiosity, but nothing to hold up merging.  i'll defer to @antonym or merge this myself in 24 hours.

File: gating/periodic/run_system_tests.sh
Comment (johnduarte) [reply gating/periodic/run_system_tests.sh]: Brain fart on my part. Thanks for pointing out the lack of a `popd`. I have reverted this change since it is unnecessary.
Thanks @coreywright

**Comment by coreywright** [comment]
recheck_xenial_mnaio_3ctlr_2comp_3ceph_osp_13_deploy

**Review by coreywright — APPROVED** [review]

**Review by antonym — APPROVED** [review]

**Pull request merged**
